### PR TITLE
Save `TrackRecording`'s timestamps and altitudes

### PR DESCRIPTION
**Pull request by kirylkaveryn**
This PR implements writing the `timestamps` and `altitudes` during the track recording.
It continues the https://github.com/organicmaps/organicmaps/pull/9140

### CODE REVIEW

File: map/bookmark_manager.cpp
Comment (biodranik) [inline map/bookmark_manager.cpp:1153]: There is no need to introduce temporary variables here (and do unnecessary move). Members of `kml::MultiGeometry geometry;` below can be used directly.

Comment (flomnes) [reply map/bookmark_manager.cpp:1153]: As long as there are no early `throw`, `return` statements, I agree.

File: map/bookmark_helpers.cpp
Comment (flomnes) [inline map/bookmark_helpers.cpp:250]: You're adding new code to existing function `ValidateKmlData`. For each element of `data.m_tracksData`, you eliminate points that are too close to each other.

I suggest creating a new function `RemovePointsTooCloseFromLine` for this operation, if possible. You might even add a parameter for the minimal distance (instead of `kMwmPointAccuracy`).

This would allow you to add unit testing for this function.

**Review by flomnes — COMMENTED** [review]
General remark : if a geometry is described by many points all very close to each other (`distance<kMwmPointAccuracy`), you may not want to eliminate all these points, since this would lead to an empty description (new bug).

File: data/kml_test_data/generated_mixed_tracks.kml
Comment (flomnes) [inline data/kml_test_data/generated_mixed_tracks.kml:133]: This URL doesn't work on my machine ❌

File: map/bookmark_manager.cpp
Comment (kirylkaveryn) [reply map/bookmark_manager.cpp]: Fixed!

File: map/bookmark_helpers.cpp
Comment (kirylkaveryn) [reply map/bookmark_helpers.cpp:250]: Thanks! I'll add this change as part of the #9140

File: data/kml_test_data/generated_mixed_tracks.kml
Comment (kirylkaveryn) [reply data/kml_test_data/generated_mixed_tracks.kml:133]: It seems that this image link is stub only for the testing purposes

**Review by biodranik — COMMENTED** [review]
Thanks! This one is good to merge after merging and rebasing on top of https://github.com/organicmaps/organicmaps/pull/9140

**Review by biodranik — APPROVED** [review]
Thanks!

**Pull request merged**
